### ISSUE TITLE
transformClassesWithAndroidEntryPointTransformFor is not cacheable

### ISSUE THREAD

**Issue by tinder-inakivillar** [google/dagger#2480]
Hi, we are using hilt 2.33-beta. We have noticed in CI and local builds tasks type `transformClassesWithAndroidEntryPointTransformFor` are never cached. 
We have tested scenarios on the same commit for CI and the result is:
```
The task was not up-to-date because of the following reasons:	
Output property 'streamOutputFolder' file app/build/intermediates/transforms/AndroidEntryPointTransform/debug has been removed.	
Output property 'streamOutputFolder' file app/build/intermediates/transforms/AndroidEntryPointTransform/debug/0.jar has been removed.	
Output property 'streamOutputFolder' file app/build/intermediates/transforms/AndroidEntryPointTransform/debug/1 has been removed.	
```
This is happening for tasks in modules using Hilt. 

Thanks

**Comment by danysantiago** [comment]
What version of AGP are you on? This should be fixed when using AGP 4.2.0+

**Comment by tinder-inakivillar** [comment]
thanks @danysantiago,
It makes sense, we are using 4.1.2. I will test with 4.2.0+
